.github/workflows/build.yml

```diff
@@ -0,0 +1,29 @@
+name: build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19
+
+      - name: Build
+        run: |
+          cd ./cmd/mnemonikey
+          go build -v ./...
+
+      - name: Test
+        run: go test -v ./...
```

Comment:
Could you please leave the existing `test.yml` action as it was, and remove this `build.yml` action? I think i'd like to keep the testing pipeline we already had. 

Although i do like your change here:

```yml
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

I agree that is probably the right approach. Could you please apply that to `test.yml` as a drive-by? :smile:

Reply:
Sure, I'll remove the `build` action and apply the above changes in `test` action.